api/v1beta1/network_disruption.go

```diff
@@ -458,3 +454,46 @@ func (h NetworkDisruptionHostSpec) Validate() error {
 
 	return nil
 }
+
+func (s NetworkDisruptionServiceSpec) ExtractAffectedPortsInServicePorts(k8sService *v1.Service) ([]v1.ServicePort, []NetworkDisruptionServicePortSpec) {
+	if len(s.Ports) == 0 {
+		return k8sService.Spec.Ports, nil
+	}
+
+	servicePortsDic := map[string]v1.ServicePort{}
+	goodPorts, notFoundPorts := []v1.ServicePort{}, []NetworkDisruptionServicePortSpec{}
+
+	// Convert service ports from found k8s service to a dictionary in order to facilitate the filtering of the ports
+	for _, port := range k8sService.Spec.Ports {
+		servicePortsDic[strconv.Itoa(int(port.Port))] = port
+		if port.Name != "" {
+			servicePortsDic[port.Name] = port
+		}
+	}
+
+	for _, allowedPort := range s.Ports {
+		if allowedPort.Port != 0 {
+			servicePort := servicePortsDic[strconv.Itoa(allowedPort.Port)]
+
+			if servicePort.Port == 0 || (allowedPort.Name != "" && allowedPort.Name != servicePort.Name) {
+				notFoundPorts = append(notFoundPorts, allowedPort)
+
+				continue
+			}
+
+			goodPorts = append(goodPorts, servicePort)
+		} else if allowedPort.Name != "" {
+			servicePort := servicePortsDic[allowedPort.Name]
+
+			if servicePort.Port == 0 || (allowedPort.Port != 0 && servicePort.Port == int32(allowedPort.Port)) {
+				notFoundPorts = append(notFoundPorts, allowedPort)
+
+				continue
+			}
```

Comment:
```suggestion
			servicePort := servicePortsDic[fmt.Sprintf("name-%d", allowedPort.Name)]

			if !ok {
				notFoundPorts = append(notFoundPorts, allowedPort)

				continue
			}
```

`allowedPort != 0` can't happen because part of first if branch.